src/main/java/ca/on/oicr/gsi/dimsum/service/filtering/CaseFilterKey.java

```diff
@@ -137,4 +152,24 @@ private static CompletedGate getGate(String label) {
     return gate;
   }
 
+  private static LocalDate getCompletionDate(Case x) {
+    List<CaseDeliverable> deliverables = x.getDeliverables();
+    if (deliverables.isEmpty()) {
+      return null;
+    }
+    List<CaseRelease> releases = deliverables.stream()
+        .flatMap(deliverable -> deliverable.getReleases().stream())
+        .collect(Collectors.toList());
+    if (releases.isEmpty()) {
+      return null;
+    }
+    if (releases.stream()
+        .anyMatch(release -> release.getQcPassed() == null || !release.getQcPassed())) {
+      return null;
+    }
+    return releases.stream()
+        .map(CaseRelease::getQcDate)
+        .max(LocalDate::compareTo)
+        .orElse(null);
+  }
```

Comment:
The exact same method exists in `CaseTatReport`. Move this to `SampleUtils` and use for both (In the report, it takes `RowData` as a parameter, but this can easily bechanged to a `Case` instead like here.

Maybe rename `SampleUtils` to `DataUtils` so it makes more sense to include utility methods for samples, cases, or any other data